### PR TITLE
fix(vpc-routing-table): support removing of  advertise routes and accept routes from array

### LINKED ISSUE #5036: ibm_is_vpc_routing_table removing all entries to the advertise_routes_to list had no effect
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

```
(vpc-transit) vpc-transit/firewall_tf % tf version
Terraform v1.5.3
on darwin_arm64
+ provider registry.terraform.io/ibm-cloud/ibm v1.61.0

Your version of Terraform is out of date! The latest version
is 1.6.6. You can update by downloading from https://www.terraform.io/downloads.html
```
### Affected Resource(s)

* ibm_is_vpc_routing_table

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "ibm_is_vpc_routing_table" "transit_tgw_ingress" {
  vpc                           = local.transit_vpc.id
  name                          = "tgw-ingress"
  route_direct_link_ingress     = false
  route_transit_gateway_ingress = true
  route_vpc_zone_ingress        = false
  # todo correct:
  # advertise_routes_to           = ["transit_gateway"]
  # todo test:
  advertise_routes_to = []
}
```

### Debug Output

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.firewall[0].ibm_is_vpc_routing_table.transit_tgw_ingress will be updated in-place
  ~ resource "ibm_is_vpc_routing_table" "transit_tgw_ingress" {
      ~ advertise_routes_to              = [
          - "transit_gateway",
        ]
        id                               = "r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e"
        name                             = "tgw-ingress"
        # (13 unchanged attributes hidden)
    }

  # module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["0"] will be updated in-place
  ~ resource "ibm_is_vpc_routing_table_route" "transit_tgw_ingress" {
      ~ advertise       = true -> false
        id              = "r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-d5674bdc-cd80-42a4-872c-19fd0395a4fe"
        name            = "zus-south-1-to-enterprise-0"
        # (13 unchanged attributes hidden)
    }

  # module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["1"] will be updated in-place
  ~ resource "ibm_is_vpc_routing_table_route" "transit_tgw_ingress" {
      ~ advertise       = true -> false
        id              = "r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-8cfdfcae-2fba-45ed-a98b-9f66b043adce"
        name            = "z0-entire-zone-1"
        # (13 unchanged attributes hidden)
    }

  # module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0] will be updated in-place
  ~ resource "ibm_is_lb_pool" "zone" {
        id                       = "r006-c521bc4c-5b48-451b-bbc5-305a4fee3125/r006-361ae609-6e7c-414f-a8fc-8cce4862c32e"
        name                     = "aaa-fw-z1-s3"
      + session_persistence_type = "source_ip"
        # (12 unchanged attributes hidden)
    }

Plan: 0 to add, 4 to change, 0 to destroy.


  Enter a value: yes

module.firewall[0].ibm_is_vpc_routing_table.transit_tgw_ingress: Modifying... [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee3125/r006-361ae609-6e7c-414f-a8fc-8cce4862c32e]
module.firewall[0].ibm_is_vpc_routing_table.transit_tgw_ingress: Modifications complete after 7s [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 10s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 20s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 30s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 40s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 50s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 1m0s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 1m10s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 1m20s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Still modifying... [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee31...6-361ae609-6e7c-414f-a8fc-8cce4862c32e, 1m30s elapsed]
module.firewall[0].module.transit_zones["0"].ibm_is_lb_pool.zone[0]: Modifications complete after 1m34s [id=r006-c521bc4c-5b48-451b-bbc5-305a4fee3125/r006-361ae609-6e7c-414f-a8fc-8cce4862c32e]
module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["1"]: Modifying... [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-8cfdfcae-2fba-45ed-a98b-9f66b043adce]
module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["0"]: Modifying... [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-d5674bdc-cd80-42a4-872c-19fd0395a4fe]
module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["1"]: Modifications complete after 1s [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-8cfdfcae-2fba-45ed-a98b-9f66b043adce]
module.firewall[0].ibm_is_vpc_routing_table_route.transit_tgw_ingress["0"]: Modifications complete after 1s [id=r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387/r006-03b926a8-1581-4806-858a-cf463d89063e/r006-d5674bdc-cd80-42a4-872c-19fd0395a4fe]

Apply complete! Resources: 0 added, 4 changed, 0 destroyed.

```

End result was no change in the cloud routing table:

```
(vpc-transit) vpc-transit/firewall_tf % ic is vpc-routing-table r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387 tgw-ingress
Getting routing table tgw-ingress of vpc r006-994cdbaf-5d79-478f-94f2-1d1ea5b41387 under account Powell Quiring's Account as user pquiring@us.ibm.com...

ID                                         r006-03b926a8-1581-4806-858a-cf463d89063e
Name                                       tgw-ingress
Status                                     stable
Accept routes from resource type filters   -
Route Direct Link Ingress                  false
Route Internet Ingress                     false
Route Transit Gateway Ingress              true
Route VPC Zones Ingress                    false
Routes                                     ID                                          Name                          Action    Status   Destination      Next hop      Zone         Priority   Origin
                                           r006-8cfdfcae-2fba-45ed-a98b-9f66b043adce   z0-entire-zone-1              deliver   stable   10.1.0.0/16      10.1.15.199   us-south-1   2          user
                                           r006-d5674bdc-cd80-42a4-872c-19fd0395a4fe   zus-south-1-to-enterprise-0   deliver   stable   192.168.0.0/16   10.1.15.199   us-south-1   2          user

Advertise routes to                        transit_gateway
Subnets                                    ID                          Name
                                           No subnets were attached.

Created                                    2024-01-09T06:38:22-08:00
```

### Expected Behavior

advertise routes to should have changed.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #5036

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by deepaksibm** [comment]
```







tfpt TestAccIBMISVPCRoutingTable_advertiseRoutesTO
=== RUN   TestAccIBMISVPCRoutingTable_advertiseRoutesTO
--- PASS: TestAccIBMISVPCRoutingTable_advertiseRoutesTO (136.69s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     137.869s

tfpt TestAccIBMISVPCRoutingTable_acceptRoutesFrom
=== RUN   TestAccIBMISVPCRoutingTable_acceptRoutesFrom
--- PASS: TestAccIBMISVPCRoutingTable_acceptRoutesFrom (91.62s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     92.791s
```

**Comment by deepaksibm** [comment]
> I do not see any test changes. Are there tests for this feature?

@powellquiring Test cases were already there for this feature. And some cases are now enhanced to capture this scenario.

File: ibm/service/vpc/resource_ibm_is_vpc_routing_table_test.go [hunk 1]
```diff
@@ -206,3 +224,50 @@ resource "ibm_is_vpc_routing_table" "test_ibm_is_vpc_routing_table" {
...
+resource "ibm_is_vpc" "testacc_vpc" {
+	name = "%s"
+}
+resource "ibm_is_vpc_routing_table" "test_ibm_is_vpc_routing_table" {
+	depends_on = [ibm_is_vpc.testacc_vpc]
+	route_direct_link_ingress = true
+	vpc = ibm_is_vpc.testacc_vpc.id
+	name = "%s"
+	accept_routes_from_resource_type=[]
+	advertise_routes_to=[]
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_vpc_routing_table_test.go:271]: We need to set explictly to empty array ? what happens if user remove the argument itself from tf file advertise_routes_to

Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_vpc_routing_table_test.go:271]: It is a optional and computed field. so if user removes the argument itself, then there won't be any change shown. 
user has to explicitly set empty array.

**Pull request merged**